### PR TITLE
[PIRK-19] Further restructuring in schema loading.

### PR DESCRIPTION
**Pull request by tellison**
- Additional re-factoring of the query schema loading code.
 - Pull out XML document parsing to helper methods.
 - Narrowing of exception type thrown by data partitioner methods.

### CODE REVIEW

File: src/main/java/org/apache/pirk/schema/data/DataSchemaLoader.java [hunk 1]
```diff
@@ -181,19 +172,34 @@ public DataSchema loadSchema(InputStream stream) throws IOException, PIRExceptio
...
     return dataSchema;
   }
 
-  private void parseElementNode(Element eElement, DataSchema schema) throws PIRException
+  private Document parseXMLDocument(InputStream stream) throws IOException, PIRException
+  {
+    Document doc;
+    try
+    {
+      DocumentBuilder dBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
```
Comment (smarthi) [inline src/main/java/org/apache/pirk/schema/data/DataSchemaLoader.java]: can this be made try (DocumentBuilder .....) ?

**Comment by ellisonanne** [comment]
+1

**Comment by ellisonanne** [comment]
Are you ready for this to be merged?

File: src/main/java/org/apache/pirk/schema/data/DataSchemaLoader.java (hunk 1 above)
Comment (tellison) [inline src/main/java/org/apache/pirk/schema/data/DataSchemaLoader.java]: No, the DocumentBuilder does not need to (cannot) be closed.

**Comment by tellison** [comment]
@ellisonanne yes, this is ready to merge pls.

There may be minor changes but this is the bulk of the clean-up restructuring I intend to do right now.

**Pull request closed without merging**